### PR TITLE
Dev: ROS: Split into ROS 1 and ROS 2 docs

### PR DESCRIPTION
**Pull request by Ryanf55**
Before and After:

image

I want to also recommend using ROS 2. Note the ethernet changes were because it wasn't representing in the TOC correctly. Incorrect headings were used.

### CODE REVIEW

File: dev/source/docs/ros.rst [hunk 1]
```diff
@@ -1,29 +1,35 @@
 .. _ros:
 
-========
-ROS/ROS2
-========
+=========
+ROS 1 / ROS 2
+=========
```
Comment (Hwurzburg) [inline dev/source/docs/ros.rst:5]: underline/overline error

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by Ryanf55** [comment]
What do you think about moving up the `Prerequisites` section? We get a fair number of people trying to build ardupilot in a ros2 workspace before they even know what ArduPilot is or how to use ROS 2.

**Comment by Hwurzburg** [comment]
I like it....first thing before ROS1/2 is even mentioned....

**Comment by Ryanf55** [comment]
> I like it....first thing before ROS1/2 is even mentioned....

Done, it's nearly at the top. I think it's work introducing ROS 1 and ROS 2, otherwise the links in the pre-reqs don't make sense.

**Pull request merged**